File: app/build_old_105631/tmp/kapt3/stubs/debug/com/example/wardeobe/data/WardrobeRepository.java
package com.example.wardeobe.data;

@kotlin.Metadata(mv = {1, 9, 0}, k = 1, xi = 48, d1 = {"\u0000F\n\u0002\u0018\u0002\n\u0002\u0010\u0000\n\u0000\n\u0002\u0018\u0002\n\u0000\n\u0002\u0018\u0002\n\u0002\b\u0002\n\u0002\u0010\u0002\n\u0000\n\u0002\u0010\u000e\n\u0002\b\u0005\n\u0002\u0010 \n\u0002\u0018\u0002\n\u0002\b\u0005\n\u0002\u0010$\n\u0000\n\u0002\u0018\u0002\n\u0000\n\u0002\u0018\u0002\n\u0002\b\u0002\b\u0007\u0018\u00002\u00020\u0001B\u0015\u0012\u0006\u0010\u0002\u001a\u00020\u0003\u0012\u0006\u0010\u0004\u001a\u00020\u0005\u00a2\u0006\u0002\u0010\u0006J\u001e\u0010\u0007\u001a\u00020\b2\u0006\u0010\t\u001a\u00020\n2\u0006\u0010\u000b\u001a\u00020\nH\u0086@\u00a2\u0006\u0002\u0010\fJ\u0016\u0010\r\u001a\u00020\b2\u0006\u0010\u000b\u001a\u00020\nH\u0086@\u00a2\u0006\u0002\u0010\u000eJ\u001c\u0010\u000f\u001a\b\u0012\u0004\u0012\u00020\u00110\u00102\u0006\u0010\t\u001a\u00020\nH\u0086@\u00a2\u0006\u0002\u0010\u000eJ&\u0010\u0012\u001a\u00020\u00112\u0006\u0010\t\u001a\u00020\n2\u0006\u0010\u0013\u001a\u00020\n2\u0006\u0010\u0014\u001a\u00020\nH\u0086@\u00a2\u0006\u0002\u0010\u0015J(\u0010\u0016\u001a\f\u0012\u0002\b\u0003\u0012\u0002\b\u0003\u0018\u00010\u00172\u0006\u0010\u0018\u001a\u00020\u00192\u0006\u0010\u001a\u001a\u00020\u001bH\u0086@\u00a2\u0006\u0002\u0010\u001cR\u000e\u0010\u0004\u001a\u00020\u0005X\u0082\u0004\u00a2\u0006\u0002\n\u0000R\u000e\u0010\u0002\u001a\u00020\u0003X\u0082\u0004\u00a2\u0006\u0002\n\u0000\u00a8\u0006\u001d"}, d2 = {"Lcom/example/wardeobe/data/WardrobeRepository;", "", "functions", "Lcom/google/firebase/functions/FirebaseFunctions;", "firestore", "Lcom/google/firebase/firestore/FirebaseFirestore;", "(Lcom/google/firebase/functions/FirebaseFunctions;Lcom/google/firebase/firestore/FirebaseFirestore;)V", "deleteItem", "", "uid", "", "publicId", "(Ljava/lang/String;Ljava/lang/String;Lkotlin/coroutines/Continuation;)Ljava/lang/Object;", "deleteTemporaryGarment", "(Ljava/lang/String;Lkotlin/coroutines/Continuation;)Ljava/lang/Object;", "fetchItems", "", "Lcom/example/wardeobe/model/ClothingItem;", "uploadItem", "base64Image", "category", "(Ljava/lang/String;Ljava/lang/String;Ljava/lang/String;Lkotlin/coroutines/Continuation;)Ljava/lang/Object;", "uploadTemporaryGarment", "", "context", "Landroid/content/Context;", "uri", "Landroid/net/Uri;", "(Landroid/content/Context;Landroid/net/Uri;Lkotlin/coroutines/Continuation;)Ljava/lang/Object;", "app_debug"})
public final class WardrobeRepository {
    @org.jetbrains.annotations.NotNull()
    private final com.google.firebase.functions.FirebaseFunctions functions = null;
    @org.jetbrains.annotations.NotNull()
    private final com.google.firebase.firestore.FirebaseFirestore firestore = null;
    
    public WardrobeRepository(@org.jetbrains.annotations.NotNull()
    com.google.firebase.functions.FirebaseFunctions functions, @org.jetbrains.annotations.NotNull()
    com.google.firebase.firestore.FirebaseFirestore firestore) {
        super();
    }
    
    /**
     * Calls a Cloud Function to fetch wardrobe items for the given user id.
     */
    @org.jetbrains.annotations.Nullable()
    public final java.lang.Object fetchItems(@org.jetbrains.annotations.NotNull()
    java.lang.String uid, @org.jetbrains.annotations.NotNull()
    kotlin.coroutines.Continuation<? super java.util.List<com.example.wardeobe.model.ClothingItem>> $completion) {
        return null;
    }
    
    /**
     * Calls a Cloud Function to delete a clothing item.
     */
    @org.jetbrains.annotations.Nullable()
    public final java.lang.Object deleteItem(@org.jetbrains.annotations.NotNull()
    java.lang.String uid, @org.jetbrains.annotations.NotNull()
    java.lang.String publicId, @org.jetbrains.annotations.NotNull()
    kotlin.coroutines.Continuation<? super kotlin.Unit> $completion) {
        return null;
    }
    
    /**
     * Uploads a new clothing item via a Cloud Function. The function returns the created ClothingItem.
     */
    @org.jetbrains.annotations.Nullable()
    public final java.lang.Object uploadItem(@org.jetbrains.annotations.NotNull()
    java.lang.String uid, @org.jetbrains.annotations.NotNull()
    java.lang.String base64Image, @org.jetbrains.annotations.NotNull()
    java.lang.String category, @org.jetbrains.annotations.NotNull()
    kotlin.coroutines.Continuation<? super com.example.wardeobe.model.ClothingItem> $completion) {
        return null;
    }
    
    /**
     * Uploads a local image URI temporarily to Cloudinary for VTO.
     */
    @org.jetbrains.annotations.Nullable()
    public final java.lang.Object uploadTemporaryGarment(@org.jetbrains.annotations.NotNull()
    android.content.Context context, @org.jetbrains.annotations.NotNull()
    android.net.Uri uri, @org.jetbrains.annotations.NotNull()
    kotlin.coroutines.Continuation<? super java.util.Map<?, ?>> $completion) {
        return null;
    }
    
    /**
     * Deletes a temporary garment image from Cloudinary.
     * Used by OutfitViewModel after VTO generation.
     */
    @org.jetbrains.annotations.Nullable()
    public final java.lang.Object deleteTemporaryGarment(@org.jetbrains.annotations.NotNull()
    java.lang.String publicId, @org.jetbrains.annotations.NotNull()
    kotlin.coroutines.Continuation<? super kotlin.Unit> $completion) {
        return null;
    }
}

File: app/src/main/java/com/example/wardeobe/data/WardrobeRepository.kt
package com.example.wardeobe.data

import com.google.firebase.functions.FirebaseFunctions
import kotlinx.coroutines.CancellationException
import com.google.firebase.firestore.FirebaseFirestore
import com.example.wardeobe.model.ClothingItem
import kotlinx.coroutines.tasks.await
import com.cloudinary.Cloudinary
import com.cloudinary.utils.ObjectUtils
import com.example.wardeobe.BuildConfig
import android.util.Log

class WardrobeRepository(
    private val functions: FirebaseFunctions,
    private val firestore: FirebaseFirestore
) {
    /**
     * Calls a Cloud Function to fetch wardrobe items for the given user id.
     */
    suspend fun fetchItems(uid: String): List<ClothingItem> {
        val data = hashMapOf("uid" to uid)
        val result = functions
            .getHttpsCallable("fetchWardrobeItems")
            .call(data)
            .await()
        @Suppress("UNCHECKED_CAST")
        val items = (result.data as? List<Map<String, Any>>).orEmpty()
        return items.map { ClothingItem.fromMap(it) }
    }

    /**
     * Calls a Cloud Function to delete a clothing item.
     */
    suspend fun deleteItem(uid: String, publicId: String) {
        val data = hashMapOf("uid" to uid, "publicId" to publicId)
        functions
            .getHttpsCallable("deleteWardrobeItem")
            .call(data)
            .await()
    }

    /**
     * Uploads a new clothing item via a Cloud Function. The function returns the created ClothingItem.
     */
    suspend fun uploadItem(uid: String, base64Image: String, category: String): ClothingItem {
        val data = hashMapOf(
            "uid" to uid,
            "imageBase64" to base64Image,
            "category" to category
        )
        val result = functions
            .getHttpsCallable("uploadWardrobeItem")
            .call(data)
            .await()
        @Suppress("UNCHECKED_CAST")
        val map = result.data as Map<String, Any>
        return ClothingItem.fromMap(map)
    }
    /**
     * Uploads a local image URI temporarily via a Cloud Function for VTO.
     */
    suspend fun uploadTemporaryGarment(context: android.content.Context, uri: android.net.Uri): Map<*, *>? {
        return kotlinx.coroutines.withContext(kotlinx.coroutines.Dispatchers.IO) {
            try {
                val compressed = com.example.wardeobe.util.ImageCompressor.compressImage(context, uri) ?: return@withContext null
                val base64Image = android.util.Base64.encodeToString(compressed, android.util.Base64.NO_WRAP)

                val data = hashMapOf(
                    "imageBase64" to base64Image
                )

                val result = functions
                    .getHttpsCallable("uploadTemporaryGarment")
                    .call(data)
                    .await()
                
                result.data as? Map<*, *>
            } catch (e: CancellationException) { throw e } catch (e: Exception) {
                Log.e("WardrobeRepository", "Temporary upload failed: ${e.message}", e)
                null
            }
        }
    }

    /**
     * Deletes a temporary garment image via a Cloud Function.
     * Used by OutfitViewModel after VTO generation.
     */
    suspend fun deleteTemporaryGarment(publicId: String) {
        try {
            val data = hashMapOf("publicId" to publicId)
            functions
                .getHttpsCallable("deleteTemporaryGarment")
                .call(data)
                .await()
            Log.d("WardrobeRepository", "Deleted temporary garment: $publicId")
        } catch (e: CancellationException) { throw e } catch (e: Exception) {
            Log.e("WardrobeRepository", "Failed to delete temporary garment: $publicId", e)
        }
    }
}
